lib/foreman_maintain/package_manager/apt.rb

```diff
@@ -41,6 +41,10 @@ def check_update(packages: nil, with_status: false)
       apt_action('upgrade --dry-run', packages, :with_status => with_status)
     end
 
+    def update_available?(package)
+      sys.execute!('apt-get upgrade --dry-run &> /dev/null').include?(package)
```

Comment:
`&> /dev/null` drops all output, so, uh, this will never be true? ;-)

That said, no, there is no *good* alternative for that.

You could check the output of `apt-cache policy <package>` and check if `Installed` and `Candidate` are the same.
Or you could check the output of `apt-get install <package> --dry-run` to contain a line like `Inst <package> …`

The problem with your approach is, if there are two packages `fun` and `well-this-isnt-funny`, your `update_available?('fun')` would also trigger if the available update is actually for `well-this-isnt-funny`.

Reply:
> `&> /dev/null` drops all output, so, uh, this will never be true? ;-)
> 
> That said, no, there is no _good_ alternative for that.
> 
> You could check the output of `apt-cache policy <package>` and check if `Installed` and `Candidate` are the same. Or you could check the output of `apt-get install <package> --dry-run` to contain a line like `Inst <package> …`
> 
> The problem with your approach is, if there are two packages `fun` and `well-this-isnt-funny`, your `update_available?('fun')` would also trigger if the available update is actually for `well-this-isnt-funny`.

Yeah I also thought there command won't return anything, but that was not the case,
~~~
2] pry(#<ForemanMaintain::PackageManager::Apt>)> sys.execute!("apt-get upgrade --dry-run &> /dev/null")
=> "Reading package lists...\nBuilding dependency tree...\nReading state information...\nCalculating upgrade...\nThe following packages have been kept back:\n  linux-cloud-tools-generic linux-cloud-tools-virtual linux-generic linux-headers-generic linux-image-generic\nThe following packages will be upgraded:\n  base-files language-pack-en language-
~~~

I am checking on the `apt-cache` and `dry-run` and will update the code accordingly.

Reply:
I'd still omit the `&>/dev/null` then, it's confusing. (I recall it can be a bashism, and Debian doesn't use Bash by default)

Reply:
> I'd still omit the `&>/dev/null` then, it's confusing. (I recall it can be a bashism, and Debian doesn't use Bash by default)

Agree, i just wanted to hide foreman-maintain printing whole command output on terminal and I can't see anything like `-q` flag which we have with yum. So lets just print the output on terminal while checking if updates for package is available? or maybe I should be using shell commands to trim the output and check it instead of ruby which will help a bit.

Reply:
Why not using
```ruby
output, status = Open3.capture2('apt-get …')
if status.success? && output.includes?('search')
  …
end
```